allensdk/brain_observatory/behavior/behavior_project_cache.py

```diff
@@ -0,0 +1,133 @@
+import pandas as pd
+from allensdk.api.cache import Cache
+from allensdk.brain_observatory.ecephys.ecephys_project_cache import call_caching
+from allensdk.brain_observatory.ecephys.file_promise import write_from_stream
+from allensdk.brain_observatory.behavior.behavior_project_api.behavior_project_fixed_api import BehaviorProjectFixedApi
+from allensdk.brain_observatory.behavior.behavior_project_api.behavior_project_lims_api import BehaviorProjectLimsApi
+from allensdk.brain_observatory.behavior.behavior_ophys_api.behavior_ophys_nwb_api import BehaviorOphysNwbApi, ExtendedBehaviorOphysNwbApi
+from allensdk.brain_observatory.behavior.behavior_ophys_session import BehaviorOphysSession, ExtendedBehaviorOphysSession
+
+csv_io = {
+    'reader': lambda path: pd.read_csv(path, index_col='Unnamed: 0'),
+    'writer': lambda path, df: df.to_csv(path)
+}
+
+class BehaviorProjectCache(Cache):
+
+    SESSIONS_KEY = 'sessions'
+
+    NWB_DIR_KEY = 'nwb_files'
+    SESSION_NWB_KEY = 'session_nwb'
+
+    ANALYSIS_FILES_DIR_KEY = 'analysis_files'
+    TRIAL_RESPONSE_KEY = 'trial_response'
+    FLASH_RESPONSE_KEY = 'flash_response'
+    EXTENDED_STIM_COLUMNS_KEY = 'extended_stimulus_presentations'
+    EXPOSURE_NUMBER_KEY = 'exposure_number'
+
+    MANIFEST_VERSION = '0.0.1'
+
+    def __init__(self, fetch_api, **kwargs):
+        
+        kwargs['manifest'] = kwargs.get('manifest', 'behavior_project_manifest.json')
+        kwargs['version'] = kwargs.get('version', self.MANIFEST_VERSION)
+
+        super(BehaviorProjectCache, self).__init__(**kwargs)
+        self.fetch_api = fetch_api
+
+    def get_sessions(self):
+        path = self.get_cache_path(None, self.SESSIONS_KEY)
+        return call_caching(self.fetch_api.get_sessions, path=path, strategy='lazy', writer=csv_io["writer"], reader=csv_io['reader'])
+
+    def get_session_data(self, experiment_id):
+        # TODO: Use session ID here? will need to specify per-plane if we want to support mesoscope though, which would be experiment ID
```

Comment:
I vote session. My understanding is that an experiment corresponds to an imaging plane and a session has 1+ experiments.